nemo_aligner/utils/trt_llm.py

```diff
@@ -158,7 +185,7 @@ def generate(self, inputs):
             # stop strings after it has stopped. so we need to
             # be slightly inefficient and then remove the excess later on
             valid_token_lengths = valid_tokens.sum(-1, keepdims=True)
-            max_unpadded_length = max(valid_token_lengths.max(), max_length)
+            max_unpadded_length = valid_token_lengths.max()
```

Comment:
Can you just add an `assert max_length <= max_unpadded_length` to be safe?

Reply:
9254faf6fab85a3ae624a1b0b23f911abafa4bd9